mkosi

```diff
@@ -1364,13 +1367,17 @@ def enable_networkd(root: str) -> None:
     os.symlink("../run/systemd/resolve/stub-resolv.conf", os.path.join(root, "etc/resolv.conf"))
 
     with open(os.path.join(root, "etc/systemd/network/all-ethernet.network"), "w") as f:
-        f.write("""\
-[Match]
-Type=ether
+        f.write(
+            dedent(
```

Comment:
Nit but this seems like excessive newlining (we already have 5k lines as is :p). I would just do `f.write(dedent(f"""\` and end with `"""))`

Reply:
Just to be clear: 

```python
f.write(dedent(f"""\
    [Unit]
"""))
```

Reply:
I could definitely do that, but in the long run I want to convince all of you to use black for auto-formatting, since it integrates nicely into CI (thus preventing tabs vs spaces issues like we had at one point) and optimizes for small diffs, making PRs nicer to read. :)

Reply:
Maybe add a helper function then for writing and dedenting? (Can we monkey patch f.write to dedent for us? I think we're in the scenario where we always want dedent)

Reply:
I wouldn't like to monkey patch that. In the medium term I want to switch the `str` that are paths to `pathlib.Path`. I've started the work, but I've run into some problems getting this to type check and there is such an awful lot of those. This would then change
```python
foo = "somefile"
with open(os.path.join(root, foo), "w") as f:
    f.write(
        dedent(
            f"""
            content
            """
        )
    )
```
to
```python
foo = Path("somefile")
(root / foo).write_text(
    dedent(
        f"""
        content
        """
    )
)
```
Since I have my own wrapper for `Path` to wrap some Mkosi's path-related functions, I'll happily add a `.write_text_dedent()` then as well. So the discomfort should not last for too long. :)

Reply:
OK, sounds like a good plan. With your focus on code cleanup, I was already expecting a pathlib PR at some point.

Reply:
The road there is somewhat winding and #483 is part of that plan. The argparse machinery seems to be somewhat finicky when trying to get the correct types out, which is why I want to first get a proper interface between the CLI and the rest.